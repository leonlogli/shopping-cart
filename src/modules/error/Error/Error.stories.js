import React from 'react'
import Button from '@material-ui/core/Button'

import Error from './Error'

export default {
  title: 'Error',
}

export const Default = () => <Error />

export const withActions = () => (
  <Error
    actions={
      <Button variant="contained" color="primary">
        Back to home page
      </Button>
    }
  />
)

export const withStatus = () => (
  <Error
    status="403"
    actions={
      <Button variant="contained" color="primary">
        Back to home page
      </Button>
    }
  />
)

export const withTitleAndSubTitle = () => (
  <Error
    title="Error"
    subTitle="Oups!!! 500 Internal Server Error"
    actions={
      <Button variant="contained" color="primary">
        Back to home page
      </Button>
    }
  />
)
