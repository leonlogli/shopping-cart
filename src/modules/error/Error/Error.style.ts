import styled from 'styled-components'

import { Box } from '../../../components'

const ActionBox = styled(Box)`
  margin: 32px 0 0;
  text-align: center;
`

const IconImage = styled.img`
  display: block;
  width: 250px;
  margin: auto;
  margin-bottom: 24px;
`

const ErrorRoot = styled(Box)`
  padding: 48px 32px;

  > p,
  > h6 {
    text-align: center;
    font-weight: 400;
  }

  > h6 {
    line-height: 1.8;
  }
`

export { ErrorRoot, ActionBox, IconImage }
export default ErrorRoot
