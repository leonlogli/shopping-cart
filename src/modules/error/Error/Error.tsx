import React from 'react'

import { BoxProps, H6, Text } from '../../../components'

import ErrorRoot, { ActionBox, IconImage } from './Error.style'

import Icon403 from '../../../resources/svg/403.svg'
import Icon404 from '../../../resources/svg/404.svg'
import Icon500 from '../../../resources/svg/500.svg'

export type ErrorProps = BoxProps & {
  /**
   * Error status code
   * @default "500"
   */
  status?: '404' | '403' | '500'
  title?: string
  subTitle?: string
  /**
   * Operating area
   */
  actions?: React.ReactNode
}

/**
 * Used to feed back the results of a series of operational tasks that lead to an error.
 */
const Error = (props: ErrorProps) => {
  const { status = '500', title = status, subTitle, actions, ...other } = props

  const errorProps = {
    '500': {
      text: subTitle || 'Sorry, something went wrong.',
      icon: Icon500,
    },
    '403': {
      text: subTitle || 'Sorry, you are not authorized to access this page.',
      icon: Icon403,
    },
    '404': {
      text: subTitle || 'Sorry, the page you visited does not exist.',
      icon: Icon404,
    },
  }

  return (
    <ErrorRoot fullWidth {...other}>
      <IconImage src={errorProps[status].icon} alt={`Error ${status}`} />
      <H6>{title}</H6>
      <Text variant="body2" color="textSecondary">
        {errorProps[status].text}
      </Text>
      {actions && <ActionBox>{actions}</ActionBox>}
    </ErrorRoot>
  )
}

export { Error }
export default Error
