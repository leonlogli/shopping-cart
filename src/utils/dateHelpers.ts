const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
