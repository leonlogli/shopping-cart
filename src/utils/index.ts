export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
