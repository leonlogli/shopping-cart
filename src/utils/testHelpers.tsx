import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
