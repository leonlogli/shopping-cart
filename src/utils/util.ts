const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
