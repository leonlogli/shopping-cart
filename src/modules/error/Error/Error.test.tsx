import React from 'react'
import Button from '@material-ui/core/Button'

import { mountWithTheme } from '../../../utils'

import Error from './Error'

describe('<Error />', () => {
  const wrapper = mountWithTheme(
    <Error
      actions={
        <Button variant="contained" color="primary">
          Back to home page
        </Button>
      }
    />
  )

  it('should render children', () => {
    expect(wrapper.find('h6').length).toEqual(1)
    expect(wrapper.find('p').length).toEqual(1)
    expect(wrapper.find('img').length).toEqual(1)
    expect(wrapper.find('button').length).toEqual(1)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should set status', () => {
    wrapper.setProps({ status: '403' } as any)

    expect(wrapper.find('h6').text()).toEqual('403')
  })

  it('should set custom title', () => {
    const wrapper2 = mountWithTheme(<Error status="404" title="Forbidden" />)

    expect(wrapper2.find('h6').text()).toEqual('Forbidden')
  })
})
