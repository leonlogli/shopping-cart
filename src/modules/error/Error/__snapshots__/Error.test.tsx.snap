// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should match a snapshot 1`] = `
<Error
  actions={
    <WithStyles(ForwardRef(Button))
      color="primary"
      variant="contained"
    >
      Back to home page
    </WithStyles(ForwardRef(Button))>
  }
>
  <Styled(Component)
    fullWidth={true}
  >
    <ForwardRef
      className="gulRap"
      fullWidth={true}
    >
      <Styled(Styled(MuiBox))
        className="gulRap"
        fullWidth={true}
      >
        <Styled(MuiBox)
          className="eZBijs gulRap"
          fullWidth={true}
        >
          <div
            className="MuiBox-root MuiBox-root-1 eZBijs gulRap"
            fullWidth={true}
          >
            <styled.img
              alt="Error 500"
              src=""
            >
              <img
                alt="Error 500"
                className="iyCTNV"
                src=""
              />
            </styled.img>
            <H6>
              <Styled(WithStyles(ForwardRef(Typography)))
                variant="h6"
              >
                <WithStyles(ForwardRef(Typography))
                  className=""
                  variant="h6"
                >
                  <ForwardRef(Typography)
                    className=""
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    variant="h6"
                  >
                    <h6
                      className="MuiTypography-root MuiTypography-h6"
                    >
                      500
                    </h6>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </Styled(WithStyles(ForwardRef(Typography)))>
            </H6>
            <Text
              color="textSecondary"
              variant="body2"
            >
              <Styled(WithStyles(ForwardRef(Typography)))
                color="textSecondary"
                variant="body2"
              >
                <WithStyles(ForwardRef(Typography))
                  className=""
                  color="textSecondary"
                  variant="body2"
                >
                  <ForwardRef(Typography)
                    className=""
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    color="textSecondary"
                    variant="body2"
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      Sorry, something went wrong.
                    </p>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </Styled(WithStyles(ForwardRef(Typography)))>
            </Text>
            <Styled(Component)>
              <ForwardRef
                className="hhjyXT"
              >
                <Styled(Styled(MuiBox))
                  className="hhjyXT"
                >
                  <Styled(MuiBox)
                    className="dUNeVP hhjyXT"
                  >
                    <div
                      className="MuiBox-root MuiBox-root-2 dUNeVP hhjyXT"
                    >
                      <WithStyles(ForwardRef(Button))
                        color="primary"
                        variant="contained"
                      >
                        <ForwardRef(Button)
                          classes={
                            Object {
                              "colorInherit": "MuiButton-colorInherit",
                              "contained": "MuiButton-contained",
                              "containedPrimary": "MuiButton-containedPrimary",
                              "containedSecondary": "MuiButton-containedSecondary",
                              "containedSizeLarge": "MuiButton-containedSizeLarge",
                              "containedSizeSmall": "MuiButton-containedSizeSmall",
                              "disableElevation": "MuiButton-disableElevation",
                              "disabled": "Mui-disabled",
                              "endIcon": "MuiButton-endIcon",
                              "focusVisible": "Mui-focusVisible",
                              "fullWidth": "MuiButton-fullWidth",
                              "iconSizeLarge": "MuiButton-iconSizeLarge",
                              "iconSizeMedium": "MuiButton-iconSizeMedium",
                              "iconSizeSmall": "MuiButton-iconSizeSmall",
                              "label": "MuiButton-label",
                              "outlined": "MuiButton-outlined",
                              "outlinedPrimary": "MuiButton-outlinedPrimary",
                              "outlinedSecondary": "MuiButton-outlinedSecondary",
                              "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                              "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                              "root": "MuiButton-root",
                              "sizeLarge": "MuiButton-sizeLarge",
                              "sizeSmall": "MuiButton-sizeSmall",
                              "startIcon": "MuiButton-startIcon",
                              "text": "MuiButton-text",
                              "textPrimary": "MuiButton-textPrimary",
                              "textSecondary": "MuiButton-textSecondary",
                              "textSizeLarge": "MuiButton-textSizeLarge",
                              "textSizeSmall": "MuiButton-textSizeSmall",
                            }
                          }
                          color="primary"
                          variant="contained"
                        >
                          <WithStyles(ForwardRef(ButtonBase))
                            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                            component="button"
                            disabled={false}
                            focusRipple={true}
                            focusVisibleClassName="Mui-focusVisible"
                            type="button"
                          >
                            <ForwardRef(ButtonBase)
                              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                              classes={
                                Object {
                                  "disabled": "Mui-disabled",
                                  "focusVisible": "Mui-focusVisible",
                                  "root": "MuiButtonBase-root",
                                }
                              }
                              component="button"
                              disabled={false}
                              focusRipple={true}
                              focusVisibleClassName="Mui-focusVisible"
                              type="button"
                            >
                              <button
                                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                                disabled={false}
                                onBlur={[Function]}
                                onDragLeave={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseDown={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                tabIndex={0}
                                type="button"
                              >
                                <span
                                  className="MuiButton-label"
                                >
                                  Back to home page
                                </span>
                                <WithStyles(memo)
                                  center={false}
                                >
                                  <ForwardRef(TouchRipple)
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child",
                                        "childLeaving": "MuiTouchRipple-childLeaving",
                                        "childPulsate": "MuiTouchRipple-childPulsate",
                                        "ripple": "MuiTouchRipple-ripple",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                                        "root": "MuiTouchRipple-root",
                                      }
                                    }
                                  >
                                    <span
                                      className="MuiTouchRipple-root"
                                    >
                                      <TransitionGroup
                                        childFactory={[Function]}
                                        component={null}
                                        exit={true}
                                      />
                                    </span>
                                  </ForwardRef(TouchRipple)>
                                </WithStyles(memo)>
                              </button>
                            </ForwardRef(ButtonBase)>
                          </WithStyles(ForwardRef(ButtonBase))>
                        </ForwardRef(Button)>
                      </WithStyles(ForwardRef(Button))>
                    </div>
                  </Styled(MuiBox)>
                </Styled(Styled(MuiBox))>
              </ForwardRef>
            </Styled(Component)>
          </div>
        </Styled(MuiBox)>
      </Styled(Styled(MuiBox))>
    </ForwardRef>
  </Styled(Component)>
</Error>
`;
